src/riot-rs-threads/src/sync/mutex.rs

```diff
@@ -0,0 +1,205 @@
+use core::{
```

Comment:
```suggestion
#![deny(missing_docs)]                                                                                                                                       
#![deny(clippy::pedantic)]

use core::{
```

This also needs a rebase (and adapt to the ariel-os rename).

Reply:
Why are the `#[deny(...)]` statements needed here? None of the other submodules in `sync` have them.

Reply:
New code has: https://github.com/ariel-os/ariel-os/blob/3ae24d8770bc4c79923b70ee629dec9985a1275a/src/ariel-os-threads/src/sync/event.rs#L3-L4

This is basically our @ROMemories bot. It will e.g., notice that the `unsafe impl Sync` doesn't have a safety warning.

Reply:
Okay. Added them and fixed the lints.